### DISCUSSION TITLE
HeaterCooler: Fixes sensor startup bug - still have problem

### DISCUSSION THREAD

**Discussion by joddye2** [RavenSystem/esp-homekit-devices#1574] [General]
**Detailed steps to reproduce it**
I've updated to the latest (7.4) version from an older 5.x version.
After the update the DHT22 sensor shows 0.0.
Before the update the same thing happened, and if the device reboots and get 0.0 the heater starts heating.

**JSON String**
{
  "c": {
    "o": 0
  },
  "a": [
    {
      "0": {
        "r": [
          {
            "g": 15,
            "v": 0
          }
        ]
      },
      "1": {
        "r": [
          {
            "g": 15,
            "v": 0
          }
        ]
      },
      "3": {
        "r": [
          {
            "g": 15,
            "v": 1
          }
        ]
      },
      "t": 25,
      "b": [],
      "g": 2,
      "n": 2,
      "j": 3,
      "w": 1,
      "m": 15,
      "x": 30,
      "d": 0.5,
      "s": 5
    }
  ]
}

**Used device**
Wemos D1 mini

**HAA version**
7.4.0

**iOS/iPadOS version**
15

**Logs**
A .txt file with logs. Read [how to capture logs from your device](https://github.com/RavenSystem/esp-homekit-devices/wiki/Build-instructions#capture-logs)

**Screen-shots or videos showing issue**
![IMG_0605](https://user-images.githubusercontent.com/77693599/134217670-cc5538f6-6b35-4a82-a08d-dc51f1c12d06.PNG)

**Comment by Hanimaniac** [comment]
I had the same problem after reboot.
Disconnecting the sensor from the Wemos D1 and plugging it in again solved the problem.

**Comment by joddye2** [comment]
Yes that can be help or rebooting is enough sometimes. The problem is when it reboots for any reason and starts heating.
Yesterday I changed the GPIO (2->14)…now testing…

**Comment by Hanimaniac** [comment]
Maybe you have to configure the general section with a boot action like here:
https://github.com/RavenSystem/esp-homekit-devices/wiki/General-Configuration#device-actions
So that you can say, after boot, turn gpio 2 off.

**Comment by Waniaok** [comment]
I have the same problem on versions v7.3.1–> v7.6 on older versions (v2.5.1) I don't have this problem.  I am using GIPO3 (RX). helps to turn off the power. 
{"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"t":25,"s":5,"g":3 ,"d":0.50,"0":{"r":[{"g":12}]},"1":{"a":0},"3":{"r":[{"g":12,"v":1}]},"5":{"a":0}}]}

**Comment by andrewbpowers** [comment]
Same problem!
Reported weeks ago in the issues section and only got the "hint" to post it here. There were absolutely no problems with the older versions for months.
As almost always, no response from the developer.

**Comment by joddye2** [comment]
I thought that the gpio change to 14 solved my problem...but only for 22 days. Today it shows 0.0 temp again. (no manual reboot and no network connection lost and while heating) I'll give a try to disable GPIO 2 or 3 (or both).

**Reply by mariusvasss** [reply]
What sensor type? I have dht 11 on gpio 2/sonoff mini with resistor. No problem. I also have the sonoff sensor board with si7021 sensor, also no problem....

**Reply by joddye2** [reply]
Mostly with DHT22. I experience this issue with DHT11 too every time with GPIO4, if I use GPIO2 it works (mostly).

**Reply by Waniaok** [reply]
i use with this code: {"c":{"l":13,"b":[{"g":0,"t":5}]},"a":[{"t":21,"s":5,"g":14,"d":0.50,"0":{"r":[{"g":12}]},"1":{"a":0},"3":{"r":[{"g":12,"v":1}]},"5":{"a":0}}]}   firmeware 2.5.1 - no problem, work all time. 
now in new firmeware have this problem with this code.

**Comment by Waniaok** [comment]
DHT22

**Comment by mariusvasss** [comment]
Try to change it. I think si7021 is the best. It has the pull-up resistor and data sheet it's not far from dht22. 
Excuse my English...

**Comment by Waniaok** [comment]
I do not see the point in changing the sensor, it worked fine on older firmware versions

**Reply by andrewbpowers** [reply]
Correct!
The old versions worked good, the current firmware still does not work. Very annoying!
:-(

**Reply by Waniaok** [reply]
there is a chance to fix it?

**Reply by andrewbpowers** [reply]
I stopped bothering with it :-( and switched to Tasmota and a D1 Mini. Everything runs fine and without problems.

**Comment by RavenSystem** [comment]
Try v10.4.5.

**Reply by Waniaok** [reply]
ok, I test and let you know about the results

**Reply by Waniaok** [reply]
still gets 0.0 after a while :/

**Reply by Ucoz3000** [reply]
v10.4.5
Humidifier with temperature sensor
still gets 0.0 after a while :/

**Reply by Waniaok** [reply]
v10.5 still the same but I have the impression that not so often

**Reply by Waniaok** [reply]
I think I found a relationship, if the device loses wifi then the sensor shows 0.0 after recovering the connection

**Reply by andrewbpowers** [reply]
> I think I found a relationship, if the device loses wifi then the sensor shows 0.0 after recovering the connection

Thanks for the hint!
